src/envoy/http/service_control/client_cache_test.cc

```diff
@@ -268,6 +283,184 @@ TEST_F(ClientCacheQuotaResponseErrorTypeTest, ApiKeyInvalid) {
   checkAndReset(stats_base_.stats().filter_.denied_consumer_error_, 1);
 }
 
+class ClientCacheHttpRequestTest : public ClientCacheTestBase {
+ public:
+  void SetUp() override {
+    service_config_.set_service_name(kServiceName);
+    service_config_.set_service_config_id(kServiceConfigId);
+
+    cache_ = std::make_unique<ClientCache>(
+        service_config_, filter_config_, stats_base_.stats(), cm_, time_source_,
+        dispatcher_, token_fn_, token_fn_);
+
+    // Setup mock http call.
+    http_call_ = std::make_unique<MockHttpCall>();
+    check_call_factory_ = std::make_unique<MockHttpCallFactory>();
+    quota_call_factory_ = std::make_unique<MockHttpCallFactory>();
+    report_call_factory_ = std::make_unique<MockHttpCallFactory>();
+  }
+
+  void TearDown() override {
+    EXPECT_EQ(got_num_callbacks, want_num_callbacks);
+    ClientCacheTestBase::TearDown();
+  }
+
+  void injectFactoryMocks() {
+    cache_->check_call_factory_ = std::move(check_call_factory_);
+    cache_->quota_call_factory_ = std::move(quota_call_factory_);
+    cache_->report_call_factory_ = std::move(report_call_factory_);
+  }
+
+  NiceMock<Envoy::Tracing::MockSpan> mock_parent_span_;
+  std::unique_ptr<MockHttpCall> http_call_;
+
+  // Ownership of these is passed to ClientCache after calling
+  // `injectFactoryMocks`.
+  std::unique_ptr<MockHttpCallFactory> check_call_factory_;
+  std::unique_ptr<MockHttpCallFactory> quota_call_factory_;
+  std::unique_ptr<MockHttpCallFactory> report_call_factory_;
+
+  // Track the number of callbacks. This will ensure no async calls are left on
+  // destruction / tearDown.
+  int got_num_callbacks = 0;
+  int want_num_callbacks = 0;
+};
+
+class ClientCacheCheckHttpRequestTest : public ClientCacheHttpRequestTest {
+ public:
+  CheckRequest getValidCheckRequest() {
+    CheckRequest request;
+    request.set_service_name(kServiceName);
+    request.set_service_config_id(kServiceConfigId);
+    Operation* op = request.mutable_operation();
+    op->set_operation_id(kCheckOperationId);
+    op->set_operation_name("test_check_operation_name");
+    op->set_consumer_id("test-api-key");
+    return request;
+  }
+
+  CheckResponse getValidCheckResponse() {
+    CheckResponse response;
+    response.set_operation_id(kCheckOperationId);
+    response.set_service_config_id(kServiceConfigId);
+    return response;
+  }
+
+  void verifyCheckDone(const Code want_code, const Status& got_status,
+                       const CheckResponseInfo&) {
+    got_num_callbacks++;
+    EXPECT_EQ(got_status.code(), want_code)
+        << "Got message: " << got_status.message();
+  };
+};
+
+TEST_F(ClientCacheCheckHttpRequestTest, OneSuccessfulHttpCall) {
+  // Only 1 http call will be created.
```

Comment:
so this is test cache miss, that remote check call is called,  is that right?  please add comment in this test case

Reply:
Done